### ISSUE TITLE
Switching mailboxes via sidebar doesn't go to index view (from pager)

### ISSUE THREAD

**Issue by nabijaczleweli** [neomutt/neomutt#3331]
## Expected Behaviour
Switching via \<sidebar-open> to be equivalent to via \<change-folder>, and to hide pager.

## Actual Behaviour
When I have a mail open and switch mailboxes via \<sidebar-open> the mail stays open:
![image](https://user-images.githubusercontent.com/6709544/160897651-5d59fa44-373c-4277-84bd-d435a827c6f1.png)

This mail lives in Mailing lists/bugs.debian.org, and, well, as you can probably tell, it isn't from Justyna.

*Additionally*, when I press `c` I get a prompt, `+` I get an index view, and `^G` to cancel it opens the mail that's highlighted in the new mailbox (so the one from Justyna) – this is apparently #3330.

## Steps to Reproduce
1. open a mail
2. \<sidebar-next>
3. \<sidebar-open>

## How often does this happen?

- Always

## When did it start to happen?

- When I upgraded

## NeoMutt Version

Yesterday's HEAD: c6ba90d9ecea948284b898777a65f5dbb13ad369

```
NeoMutt 20211029-519-ac10a7
Copyright (C) 1996-2020 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 5.10.0-13-amd64 (x86_64)
ncurses: ncurses 6.2.20201114 (compiled with 6.2.20201114)
libidn: 1.33 (compiled with 1.33)
GPGME: 1.14.0-unknown
GnuTLS: 3.7.1
libnotmuch: 5.3.0
storage: tokyocabinet, lmdb
compression: zlib, zstd

Configure options: CC=cc LD=ld.lld --asan --mandir=/usr/share/man --libexecdir=/usr/libexec --with-mailpath=/var/mail --gpgme --lua --notmuch --gnutls --gss --idn --mixmaster --sasl --tokyocabinet --lmdb --zlib --zstd --testing --asan

Compilation CFLAGS: -g -O2 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/include -I/usr/include/lua5.4 -DNCURSES_WIDECHAR -isystem /usr/include/mit-krb5 -fsanitize=address -O0 -fno-omit-frame-pointer -fno-common

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  -autocrypt +fcntl -flock -fmemopen +futimens +getaddrinfo +gnutls +gpgme +gss 
  +hcache -homespool +idn +inotify -locales_hack +lua +mixmaster +nls +notmuch 
  -openssl +pgp +regex +sasl +smime -sqlite +sun_attachment 

Devel options:
  asan 

MAILPATH="/var/mail"
MIXMASTER="mixmaster"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

* Operating System and its version: Bullseye
* Were you using multiple copies of NeoMutt at once? no
* Were you using 'screen' or 'tmux'? no
* Is your email local (maildir) or remote (IMAP)? IMAP

**Comment by flatcap** [comment]
Thanks and sorry.
This is due to my ongoing Index/Pager changes.
There's probably a quick fix, but I'm trying to solve the problem properly.
Nearly there.

**Comment by flatcap** [comment]
Fixed by 3e1158ff6 (Index/Pager refactoring)
